src/screens/Server.js

```diff
@@ -203,6 +205,9 @@ function Server() {
       setLoading(false);
       setErrorMessage(e.message);
     }
+
+    // Forces the validation of the allowed version
+    dispatch(isVersionAllowedUpdate({ allowed: undefined }));
```

Comment:
I like this solution but can you double check that the server change flow is not affected?
For instance, we have:
- Change server from mainnet to another in mainnet
- Change server from mainnet to testnet successfully
- Change server from mainnet to testnet and fail the testnet confirm modal
- Change server from testnet to another in testnet
- Change server from testnet to mainnet

Since we are changing a variable that may change the current screen I think this is important.

Reply:
All those were working OK, but the dispatch was not being done on an optimal place.
Refactored it to inside the `executeServerChange()` method.